### PR TITLE
Add `delete` table endpoint to sdk 

### PR DESCRIPTION
**Pull request by waldruupi**
In this PR, we add support for the new table endpoints in the SDK:
- https://docs.dune.com/api-reference/tables/endpoint/delete

We also add a `_delete` function to the `BaseDuneClient`

### CODE REVIEW

File: dune_client/api/table.py [hunk 1]
```diff
@@ -99,3 +99,12 @@ def insert_table(
             data=data,
         )
         return InsertTableResult.from_dict(result_json)
+    
+    def delete_table(self, namespace: str, table_name: str) -> bool:
```
Comment (helanto) [inline dune_client/api/table.py:103]: The return type seems wrong

File: tests/e2e/test_client.py [hunk 2]
```diff
@@ -300,6 +301,27 @@ def test_insert_table_json_success(self):
                 InsertTableResult(rows_written=1),
             )
 
+    @unittest.skip("This is a plus subscription endpoint.")
```
Comment (helanto) [inline tests/e2e/test_client.py:304]: Is it ?

**Review by helanto — COMMENTED** [review]
Some comments.

File: tests/e2e/test_client.py (hunk 2 above)
Comment (waddahAldrobi) [reply tests/e2e/test_client.py:304]: You're right, I'm not sure why all of these have it, will reword

File: dune_client/api/base.py [hunk 3]
```diff
@@ -209,3 +209,14 @@ def _patch(self, route: str, params: Any) -> Any:
             timeout=self.request_timeout,
         )
         return self._handle_response(response)
+    
+    def _delete(self, route: str) -> Any:
```
[the remark this answers is not in the document]
Comment (waddahAldrobi) [reply dune_client/api/base.py:213]: See :) 
https://github.com/duneanalytics/dune-client/pull/114#discussion_r1529959462

[the remark this answers is not in the document]
Comment (helanto) [reply dune_client/api/base.py:213]: Remember discussing about it, but every time I see it 🤮

File: .gitignore [hunk 4]
```diff
@@ -7,3 +7,5 @@ _version.py
 venv/
 tmp/
 .vscode/
+build/
+.DS_Store
```
Comment (helanto) [inline .gitignore:11]: Why ?

**Review by helanto — APPROVED** [review]
A comment about `.gitignore`. Did you change on purpose ?

File: .gitignore (hunk 4 above)
Comment (waddahAldrobi) [reply .gitignore:11]: Yes, those files are noise.
I'm adding them to the ignore so that no one accidentally pushes them. 

Any concerns?

Comment (helanto) [reply .gitignore:11]: Nope! Just to make sure its not by accident

**Comment by waddahAldrobi** [comment]
@bh2smith  ugh, I forgot to squash merge again... 
Isn't there a way to make that the default behaviour?

**Comment by bh2smith** [comment]
> Isn't there a way to make that the default behaviour?

After you do it once, GitHub will use this default choice.

**Comment by waddahAldrobi** [comment]
> > Isn't there a way to make that the default behaviour?
> 
> After you do it once, GitHub will use this default choice.

Right, I was thinking of this though
https://github.com/dotnet/runtime/issues/260#issuecomment-558455365

**Pull request merged**